crates/bevy_core/src/time/time.rs

```diff
@@ -33,7 +79,12 @@ impl Time {
         self.update_with_instant(Instant::now());
     }
 
-    pub(crate) fn update_with_instant(&mut self, instant: Instant) {
+    /// Update time with a specified [`Instant`]
+    ///
+    /// This method is for use in tests, and should generally not be called on the [`Time`] resource
+    /// as part of your app. Doing so will conflict with the `time_system` called by
+    /// [`CorePlugin`](crate::CorePlugin), and result in inaccurate performance measurements.
+    pub fn update_with_instant(&mut self, instant: Instant) {
```

Comment:
```suggestion
    /// Update time with a specified [`Instant`]
    ///
    /// This method is for use in tests, and should generally not be called on the [`Time`] resource
    /// as part of your app. Doing so will conflict with the `time_system` called by
    /// [`CorePlugin`](crate::CorePlugin), and result in inaccurate performance measurements.
    #[cfg(test)]
    pub fn manual_ update_with_instant(&mut self, instant: Instant) {
        self.update_with_instant(instant);
    }
    
    pub(crate) fn update_with_instant(&mut self, instant: Instant) {
```

not sure this is a good idea, but it would make it impossible to mess with time

Reply:
TBH I don't think this actually adds much more safety. I think that aggressive doc warnings should be sufficient here.

Reply:
`#[cfg(test)]` hides the method from the new doctest. And `#[cfg(doctest)]` does not work here due to a bug in cargo (rust-lang/rust#67295). Any ideas how this could be solved?

Reply:
@mockersf, I agree with @alice-i-cecile. Is it acceptable to you to leave the patch unchanged?

Reply:
`#[cfg(test)]` only applies when testing the current crate, not when testing a dependent crate.

Reply:
It's acceptable, even more with Bjorn3 remark! (we already had this discussion on another place, and I keep forgetting that...).

It's not something I wanted to block on anyway